src/main/java/com/gfl/client/service/scenario/RestTemplateScenarioService.java

```diff
@@ -20,12 +21,15 @@
 public class RestTemplateScenarioService
         implements ScenarioService {
 
+
+
     @Value("${worker.base.uri}")
     private String baseUrl;
     @Value("${client.auth.token.value}")
     private String clientAuthToken;
 
     private final RestTemplate restTemplate;
+    private RsaManager rsaManager;
```

Comment:
add `final` to prevent NPE:
```suggestion
    private final RsaManager rsaManager;
```